samples/ProtectedMcpClient/Program.cs

```diff
@@ -134,6 +134,13 @@
 /// <param name="url">The URL to open.</param>
 static void OpenBrowser(Uri url)
 {
+    // Validate the URI scheme - only allow safe protocols
+    if (url.Scheme != Uri.UriSchemeHttp && url.Scheme != Uri.UriSchemeHttps)
```

Comment:
Would this be worth doing in the McpClient itself? We could throw during McpClient.CreateAsync rather than call the AuthorizationRedirectDelegate when given an non-HTTP or HTTPS scheme?

Reply:
We can do this as a follow up if we decide it's a good idea.